scripts/install_backend_python_libs.py

```diff
@@ -0,0 +1,364 @@
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Installation script for Oppia python backend libraries."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import collections
+import os
+import shutil
+import subprocess
+
+import python_utils
+from scripts import common
+
+import pkg_resources
+
+
+def _get_requirements_file_contents():
+    """Returns a dictionary containing all of the required library names with
+    their corresponding version strings listed in the 'requirements.txt' file.
+
+    Returns:
+        dict(string, string). Dictionary with the name of the library as the key
+        and the version string of that library as the value.
+    """
+    requirements_contents = collections.defaultdict()
+    with python_utils.open_file(
+        common.COMPILED_REQUIREMENTS_FILE_PATH, 'r') as f:
+        lines = f.readlines()
+        for l in lines:
+            l = l.strip()
+            if l.startswith('#') or len(l) == 0:
+                continue
+            library_name_and_version_string = l.split(' ')[0].split('==')
+            # Libraries are not distinguished by case so in order to make sure
```

Comment:
Ah I see, so firstly, pip doesn't distinguish libraries based on case. Libraries with differing case are treated the same so for example, 'FlaSk' is equivalent in the eyes of pip to 'flask'.
What is happening is that the listing in the requirements.txt file is auto-generated by pip-tools to be all lower case but the directories that are installed when `pip install` is run is in various case formats(depending on how it was named by the author of the library). This also affects the way we retrieve the libraries installed in the directory since those follow the format of the directory names. 

Because of the aforementioned points, when I import any names that include library names, I convert them to lower so there is no confusion if we ever need to compare them. Therefore, I added lower() to the 4 places that required them, when I parse the requirements file and directory contents (to get the specific library/versions), and in the 2 times I read from the directory since these directory/file names can be of heterogeneous case.

I don't think there's anyway to add a comparison function like @vojtechjelinek suggested since I'm using the 'in' operator to check if some of the names are in a set(). The reason I didn't change the library names to lower directly at the comparison is because it's more efficient to just change them all to lower when the relevant dictionaries/lists are created since the library names are used multiple times throughout the script.